### DISCUSSION TITLE
Should we release 0.66.3?

### DISCUSSION THREAD

**Discussion by lunaleaps** [reactwg/react-native-releases#5] [Patches]
# Released, see [discussion for 0.66.4](https://github.com/reactwg/react-native-releases/discussions/6)
Current Release: [0.66.3](https://github.com/facebook/react-native/releases/tag/v0.66.3)

Conversations on this thread are limited to 0.66 releases major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into main but did not make the 0.66.2 cut, with a link to the specific commit hash on main with the commit to cherry-pick, like this example link: facebook/react-native@bd2b7d6

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.

### List of cherry picks
1. Fix npm latest tag when running patch update which needs:
   - Adding `-v` flag to bump-oss: https://github.com/facebook/react-native/commit/b29b05fa9808dadb4cb8aab9393b8dc25d72b29
   - Extract version parsing from release script: https://github.com/facebook/react-native/commit/cc1e3abb6eb0b48b31ef8aea5ef2880a8bc1e456
   - Clean up publish-npm.js and use parseVersion: https://github.com/facebook/react-native/commit/029376c47de324e83dbe212eaf2942614512c472
   - Fix npm latest tag issue when releasing patches: https://github.com/facebook/react-native/commit/68e6788dafe1f9820003d9c921c4cef844505221
2. Revert jest preprocessor changes: https://github.com/facebook/react-native/commit/74b91c5073313cf044926d91d52c7e279cc75950
3. Renames Keyboard.removeEventListener to Keyboard.removeListener: https://github.com/facebook/react-native/commit/8880c09076e4727768ace26a74766cbe6f64021c
### Local commits to backport to main
-

**Comment by diegotsi** [comment]
Hey, I'm facing problems with Jest after upgrading from `0.65.1` to `0.66.2`, it's the same problem related [here](https://github.com/facebook/react-native/issues/32372), it's a serious issue because all unit tests are failing.

**Reply by lunaleaps** [reply]
As I mention on the 0.66.2 thread, if someone could submit a PR with a fix I'll make sure we prioritize getting it in

**Reply by rubennorte** [reply]
I'm the original author of the commit that introduced this problem in https://github.com/facebook/react-native/commit/249b8d21c4b9734bc607ed1627055f9d4c4cd447 .

The problem some people are seeing is because the `filename` option is missing when calling `babel` on that file. I can fix that easily but I think there might be other problems with this change.

Most importantly, this would enable inline requires everywhere for people using this preprocessor, which I understand wasn't the case before. Enabling inline requires in tests that didn't have it before could introduce new issues, so perhaps it's just easier to revert the change. 

Also, does someone know why people are using the preprocessor in React Native directly in their Jest configurations? I thought that was specific for React Native tests, not to use externally in other projects (they should just use the Jest preset in the `react-native` package, which relies on `metro-react-native-babel-preset` and doesn't use the preprocessor).

**Reply by diegotsi** [reply]
Hey @rubennorte that's my `jest.config` file:

```javascript
module.exports = {
  testEnvironment: 'jsdom',
  preset: 'react-native',
  moduleDirectories: ['<rootDir>/../../node_modules'],
  transformIgnorePatterns: ['../../node_modules/?!(react-native)'],
  moduleNameMapper: {
    '@xstyled/styled-components':
      '<rootDir>/../../node_modules/@xstyled/styled-components/dist/xstyled-styled-components.cjs.js',
  },
  transform: {
    '^.+\\.js$':
      '<rootDir>/../../node_modules/react-native/jest/preprocessor.js',
  },
  setupFilesAfterEnv: ['<rootDir>../../packages/_testlib/setup.js'],
  setupFiles: [
    '<rootDir>/../../node_modules/react-native-gesture-handler/jestSetup.js',
  ],
  collectCoverage: true,
  testMatch: ['<rootDir>/**/*.(spec|test).js'],
  collectCoverageFrom: ['src/**/*.{js,jsx}', '!node_modules/**'],
  coveragePathIgnorePatterns: ['/node_modules/'],
  coverageReporters: ['lcov', 'html'],
  reporters: [
    'default',
    ['jest-junit', { outputDirectory: '<rootDir>/coverage' }],
  ],
```

Removing this **transform** generate a new issue with styled-components:


![Screen Shot 2021-11-08 at 14 19 27](https://user-images.githubusercontent.com/11574033/140788299-654fa3c5-335d-4ab6-946f-2b3cc911f63c.png)


`@jsm/uikit` is a package inside our monorepo and it's working but starts failing on tests without the `transform` inside jest config.

**Reply by mark-quinn** [reply]
@rubennorte We are using `metro-react-native-babel-preset` and can confirm it still doesn't fix anything.

I personally would prefer for the change to reverted until the issue can be investigated further. It is holding back many people on upgrading their React Native version. I have no inline requires in my codebase and the only changes that I made from upgrading 0.63.4 to 0.66.1 were the changes suggested by the upgrade helper.

**Reply by jamie-phlo** [reply]
Thanks for your response @rubennorte, we are also encountering this issue and as far as I'm aware we've never specifically told jest to use the React Native preprocessor, we do as you suggest and have our jest config use the react-native preset. Our entire jest config for reference.

```
"jest": {
    "preset": "react-native",
    "setupFiles": [
      "<rootDir>/jest/setup.js"
    ],
    "setupFilesAfterEnv": [
      "@testing-library/jest-native/extend-expect"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "moduleNameMapper": {
      ".+\\.(css|styl|less|sass|scss|png|jpg|ttf|woff|woff2)$": "jest-transform-stub",
      "\\.(css|less)$": "identity-obj-proxy"
    },
    "testEnvironment": "jsdom",
    "timers": "fake",
    "reporters": [
      "default",
      "jest-junit"
    ]
  }
```

**Reply by diegotsi** [reply]
Hey Guys. I open this PR: https://github.com/facebook/react-native/pull/32559

**Reply by lunaleaps** [reply]
@diegotsi @rubennorte is reverting the change and it's landing so we probably won't need facebook/react-native#32559

**Reply by diegotsi** [reply]
Ok, @lunaleaps! let me know when the new release was published and I can close the PR. thanks!

**Reply by makenova** [reply]
> Also, does someone know why people are using the preprocessor in React Native directly in their Jest configurations? I thought that was specific for React Native tests, not to use externally in other projects (they should just use the Jest preset in the `react-native` package, which relies on `metro-react-native-babel-preset` and doesn't use the preprocessor).

This comment made me reevaluate my jest config. 
I'm maintaining an old RN app and looks like I added the FB preprocessor 3 years ago when upgrading to `0.57.1`.
Removing it does not seem to have any adverse effects and resolved an issue blocking my upgrade to `0.66.2`

Removing the FB preprocessor does not seem to be a solution for everyone.

**Reply by diegotsi** [reply]
@makenova are you using styled-components?

**Reply by makenova** [reply]
@diegotsi no, I'm not.

**Comment by lunaleaps** [comment]
Will be releasing a patch tonight or tomorrow as soon as pending (see description) changes are in

**Comment by tapz** [comment]
0.66.3 seems to be out, but the version is added to the change log as "0.64.3".

**Reply by kelset** [reply]
thanks for mentioning - I took care of it https://github.com/facebook/react-native/releases/tag/v0.66.3

**Reply by kelset** [reply]
(and PR for changelog: https://github.com/facebook/react-native/pull/32576)
